Add logic to determine hasUAVisualTransition value

rwlbuis opened the pull request:
#### 44a5e909d5d7fc8de0c7aa834670a5c5e1a3a33a
<pre>
Add logic to determine hasUAVisualTransition value
<a href="https://bugs.webkit.org/show_bug.cgi?id=264748">https://bugs.webkit.org/show_bug.cgi?id=264748</a>

Reviewed by Simon Fraser.

Add logic to determine hasUAVisualTransition value for PopStateEvent when swiping on navigation and add a test to verify this.

Rename FreezeLayerTreeDueToSwipeAnimation/UnfreezeLayerTreeDueToSwipeAnimation messages to the more generic
SwipeAnimationDidStart/swipeAnimationDidEnd, since now it needs to handle both layer tree freezing as well
as detecting the swipe on the Page for hasUAVisualTransition determination.

* LayoutTests/swipe/pushState-back-swipe-verify-ua-transition-expected.txt: Added.
* LayoutTests/swipe/pushState-back-swipe-verify-ua-transition.html: Added.
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::dispatchPopstateEvent):
* Source/WebCore/dom/PopStateEvent.h:
* Source/WebCore/page/Page.h:
(WebCore::Page::setIsInSwipeAnimation):
(WebCore::Page::isInSwipeAnimation const):
* Source/WebKit/UIProcess/ProvisionalPageProxy.cpp:
(WebKit::ProvisionalPageProxy::initializeWebPage):
(WebKit::ProvisionalPageProxy::swipeAnimationDidEnd):
(WebKit::ProvisionalPageProxy::unfreezeLayerTreeDueToSwipeAnimation): Deleted.
* Source/WebKit/UIProcess/ProvisionalPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::commitProvisionalPage):
(WebKit::WebPageProxy::navigationGestureWillEnd):
(WebKit::WebPageProxy::navigationGestureDidEnd):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::swipeAnimationDidStart):
(WebKit::WebPage::swipeAnimationDidEnd):
(WebKit::WebPage::freezeLayerTreeDueToSwipeAnimation): Deleted.
(WebKit::WebPage::unfreezeLayerTreeDueToSwipeAnimation): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: <a href="https://commits.webkit.org/275438@main">https://commits.webkit.org/275438@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3a71b7363f6b1eb66367d68c08c397397c6ed92d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41712 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20726 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44093 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44286 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37805 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/44019 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23858 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18057 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34474 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42286 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17652 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35920 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15146 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15344 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36941 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45673 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37896 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37263 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41016 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16527 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13566 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39445 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18146 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/36220 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9373 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18203 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17790 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -6000,11 +6000,13 @@ void WebPage::effectiveAppearanceDidChange(bool useDarkAppearance, bool useEleva
 void WebPage::freezeLayerTreeDueToSwipeAnimation()
 {
     freezeLayerTree(LayerTreeFreezeReason::SwipeAnimation);
+    corePage()->setIsInSwipeAnimation(true);
```
This doesn't feel like quite the right place to hook into, or we should rename this function (and the IPC) to be `swipeAnimationWillStart()` or something.

rwlbuis replied:
That makes sense, I tried to give some better naming for swipe start/end.

smfr reviewed (commented):
Does this do the right thing if the swipe results in a process-swap navigation?

rwlbuis committed:
Add logic to determine hasUAVisualTransition value
https://bugs.webkit.org/show_bug.cgi?id=264748

Reviewed by Simon Fraser.

Add logic to determine hasUAVisualTransition value for PopStateEvent when swiping on navigation and add a test to verify this.

Rename FreezeLayerTreeDueToSwipeAnimation/UnfreezeLayerTreeDueToSwipeAnimation messages to the more generic
SwipeAnimationDidStart/swipeAnimationDidEnd, since now it needs to handle both layer tree freezing as well
as detecting the swipe on the Page for hasUAVisualTransition determination.

* LayoutTests/swipe/pushState-back-swipe-verify-ua-transition-expected.txt: Added.
* LayoutTests/swipe/pushState-back-swipe-verify-ua-transition.html: Added.
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::dispatchPopstateEvent):
* Source/WebCore/dom/PopStateEvent.h:
* Source/WebCore/page/Page.h:
(WebCore::Page::setIsInSwipeAnimation):
(WebCore::Page::isInSwipeAnimation const):
* Source/WebKit/UIProcess/ProvisionalPageProxy.cpp:
(WebKit::ProvisionalPageProxy::initializeWebPage):
(WebKit::ProvisionalPageProxy::swipeAnimationDidEnd):
(WebKit::ProvisionalPageProxy::unfreezeLayerTreeDueToSwipeAnimation): Deleted.
* Source/WebKit/UIProcess/ProvisionalPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::commitProvisionalPage):
(WebKit::WebPageProxy::navigationGestureWillEnd):
(WebKit::WebPageProxy::navigationGestureDidEnd):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::swipeAnimationDidStart):
(WebKit::WebPage::swipeAnimationDidEnd):
(WebKit::WebPage::freezeLayerTreeDueToSwipeAnimation): Deleted.
(WebKit::WebPage::unfreezeLayerTreeDueToSwipeAnimation): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: https://commits.webkit.org/275438@main
